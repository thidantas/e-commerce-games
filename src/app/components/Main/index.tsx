const Main = () => (
  <main>
    <h1>Bem vindo a Main</h1>
  </main>
)

export default Main
