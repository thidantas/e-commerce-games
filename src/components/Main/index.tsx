import * as S from './styles'

export type MainProps = {
  title?: string
  description?: string
}

const Main = ({
  title = 'React Avançado',
  description = 'Typescript, React, NextJS e Styled Componts'
}: MainProps) => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="Imagem de um átomo e React Avançado" />
    <S.Title>{title}</S.Title>

    <S.Description>{description}</S.Description>

    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Um dev de frente para a tela com código"
    />
  </S.Wrapper>
)

export default Main
