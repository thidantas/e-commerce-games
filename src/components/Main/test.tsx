import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // renders component
    const { container } = render(<Main />)

    // search element and check its existence
    expect(
      screen.getByRole('heading', { name: /bem vindo a main/i })
    ).toBeInTheDocument()

    // generate snapshots

    expect(container.firstChild).toMatchSnapshot()
  })
})
