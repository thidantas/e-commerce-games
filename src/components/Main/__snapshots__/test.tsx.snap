// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-VHjGu jJdqHD"
>
  <h1>
    Bem vindo a Main
  </h1>
</main>
`;
